File: frontend/src/api/podcast.ts
import http from './http'

export interface PodcastListItem {
  id: string
  title: string
  description?: string
  duration_seconds?: number
  speaker_count: number
  style?: string
  status: string
  created_at: string
}

export interface PodcastDetail {
  id: string
  title: string
  description?: string
  original_input: string
  script_json: any
  speakers_json: any[]
  audio_file_path?: string
  duration_seconds?: number
  speaker_count: number
  style?: string
  status: string
  created_at: string
  updated_at: string
}

export function getPodcasts(page = 1, pageSize = 20): Promise<PodcastListItem[]> {
  return http.get('/podcasts', { params: { page, page_size: pageSize } })
}

export function getPodcast(id: string): Promise<PodcastDetail> {
  return http.get(`/podcasts/${id}`)
}

export function deletePodcast(id: string): Promise<void> {
  return http.delete(`/podcasts/${id}`)
}

export function getPodcastAudioUrl(id: string): string {
  return `/api/podcasts/${id}/audio`
}

File: backend/app/models/podcast.py
"""播客模型"""

import uuid
from datetime import datetime
from sqlalchemy import Column, String, Integer, Float, Text, DateTime, JSON
from app.database import Base


class Podcast(Base):
    __tablename__ = "podcasts"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    title = Column(String(200), nullable=False, comment="播客标题")
    description = Column(Text, nullable=True, comment="简介")
    original_input = Column(Text, nullable=False, comment="用户原始输入")
    script_json = Column(JSON, nullable=False, comment="完整播客脚本")
    speakers_json = Column(JSON, nullable=False, comment="角色列表")
    audio_file_path = Column(String(500), nullable=True, comment="音频文件路径")
    duration_seconds = Column(Float, nullable=True, comment="时长（秒）")
    speaker_count = Column(Integer, nullable=False, comment="角色人数")
    style = Column(String(50), nullable=True, comment="播客风格")
    status = Column(String(20), nullable=False, default="completed", comment="状态")
    created_at = Column(DateTime, default=datetime.now)
    updated_at = Column(DateTime, default=datetime.now, onupdate=datetime.now)
